### PR TITLE
CLOUD-1112 Add configure.py script to customize deployment

### PR DESCRIPTION
**Pull request by csovant**
Jira issue? CLOUD-1112
Release 6.5.0 backport required? no
6.5.0 doc changes needed? no
7.0.0 doc changes needed? no
Required README updates made? no

### CODE REVIEW

File: cicd/forgeops-tests/configure.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    os.chdir(os.path.join(os.path.abspath(os.path.dirname(__file__)), '..', '..'))
+    root_dir = os.getcwd()
+    os.chdir(current_dir)
+
+    helm_dir = os.path.join(root_dir, 'helm')
+    config_dir = os.path.join(root_dir, 'samples', 'config', TESTS_DEPLOYMENT)
+
+    env_components = "frconfig"
+    for component in TESTS_COMPONENTS.split(" "):
+        subfolder = eval('%s_HELM_SUBFOLDER' % component.upper())
```
Comment (markg58) [inline cicd/forgeops-tests/configure.py]: isn't subfolder just the component(s) passed in TESTS_COMPNENTS??

Comment (csovant) [reply cicd/forgeops-tests/configure.py]: For openam, amster, openidm and openig yes but not for the userstore, configstore and ctsstore => for these one the helm subfolder is `ds`

File: cicd/forgeops-tests/configure.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        image = eval('%s_IMAGE_NAME' % component.upper())
+        deps = eval('%s_DEPS_COMPONENTS' % component.upper())
```
Comment (andrewvinall) [inline cicd/forgeops-tests/configure.py]: `deps` is not meaningful.

File: cicd/forgeops-tests/configure.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    # save new content
+    with open(env_file, 'w') as stream:
+        try:
+            stream.write(env_content)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    print("TO DEPLOY: ../../bin/deploy.sh -n %s ../../samples/config/%s" % (TESTS_NAMESPACE, TESTS_DEPLOYMENT))
+    print("TO RUN TESTS: ./forgeops-tests.py tests/smoke -k 'test_idm and test_0_ping'")
```
Comment (andrewvinall) [inline cicd/forgeops-tests/configure.py]: Does the line above need updating?

File: cicd/forgeops-tests/configure.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        shutil.copy2('%s.orig' % env_file, env_file)
+    # save file
+    shutil.copy2(env_file, '%s.orig' % env_file)
+
+    # read content of env file
+    env_content = ""
+    with open(env_file, 'r') as stream:
+        try:
+            env_content = stream.read()
+        except yaml.YAMLError as exc:
```
Comment (andrewvinall) [inline cicd/forgeops-tests/configure.py]: Why a yaml exception for a .sh file?

File: cicd/forgeops-tests/configure.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+TESTS_COMPONENTS = "openam amster"
+if "TESTS_COMPONENTS" in os.environ:
+    TESTS_COMPONENTS = os.environ["TESTS_COMPONENTS"]
+
+IMAGE_BASE_URL = 'forgerock-docker-public.bintray.io/forgerock'
+IMAGE_PULL_POLICY = 'Always'
+
+OPENAM_HELM_SUBFOLDER = "openam"
+OPENAM_IMAGE_NAME = "openam"
+OPENAM_DEPS_COMPONENTS = "configstore"
```
Comment (andrewvinall) [inline cicd/forgeops-tests/configure.py]: What is `DEPS` an abbreviation of?

File: cicd/forgeops-tests/configure.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    TESTS_IMAGE_TAG = os.environ["TESTS_IMAGE_TAG"]
+
+TESTS_DEPLOYMENT = "smoke-deployment"
+if "TESTS_DEPLOYMENT" in os.environ:
+    TESTS_DEPLOYMENT = os.environ["TESTS_DEPLOYMENT"]
+
+TESTS_COMPONENTS = "openam amster"
+if "TESTS_COMPONENTS" in os.environ:
+    TESTS_COMPONENTS = os.environ["TESTS_COMPONENTS"]
+
```
Comment (andrewvinall) [inline cicd/forgeops-tests/configure.py]: I cannot see where the following variables are used.

File: cicd/toolbox-docker/run-smoke-tests.sh [hunk 7]
```diff
@@ -59,7 +60,7 @@ run_tests() {
     mkdir -p tmp
     rm -rf tmp/*
     cp -r reports/* tmp/
-    rm -rf reports/*
+    #rm -rf reports/*
```
[the remark this answers is not in the document]
Comment (csovant) [reply cicd/toolbox-docker/run-smoke-tests.sh]: I will remove the line

File: cicd/forgeops-tests/configure.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+TESTS_IMAGE_TAG = "7.0.0-SNAPSHOT"
+if "TESTS_IMAGE_TAG" in os.environ:
+    TESTS_IMAGE_TAG = os.environ["TESTS_IMAGE_TAG"]
+
+TESTS_DEPLOYMENT = "smoke-deployment"
+if "TESTS_DEPLOYMENT" in os.environ:
+    TESTS_DEPLOYMENT = os.environ["TESTS_DEPLOYMENT"]
+
+TESTS_COMPONENTS = "openam amster"
```
[the remark this answers is not in the document]
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Good catch

File: cicd/forgeops-tests/configure.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+OPENIG_IMAGE_NAME = "ig"
+OPENIG_DEPS_COMPONENTS = ""
+
+
+def main(argv):
+    # root_dir : computed using relative position from this file
+    current_dir = os.path.abspath(os.path.dirname(__file__))
+    os.chdir(os.path.join(os.path.abspath(os.path.dirname(__file__)), '..', '..'))
+    root_dir = os.getcwd()
+    os.chdir(current_dir)
```
[the remark this answers is not in the document]
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Yes and no :-)
With this code the `root_dir` was equal to `/tmp/forgeops/cicd/forgeops-tests/../../` but I wanted to have a "clean" path `/tmp/forgeops`

File: cicd/forgeops-tests/configure.py (hunk 2 above)
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Will rename it to dependencies

File: cicd/forgeops-tests/configure.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Done

File: cicd/forgeops-tests/configure.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+OPENIDM_HELM_SUBFOLDER = "openidm"
+OPENIDM_IMAGE_NAME = "idm"
+OPENIDM_DEPS_COMPONENTS = "postgres-openidm"
+
+OPENIG_HELM_SUBFOLDER = "openig"
+OPENIG_IMAGE_NAME = "ig"
+OPENIG_DEPS_COMPONENTS = ""
+
+
```
[the remark this answers is not in the document]
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Done

File: cicd/forgeops-tests/configure.py (hunk 3 above)
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Yes I have updated that and I have added some specific postcommit tests as discussed during the standup

File: cicd/forgeops-tests/configure.py (hunk 4 above)
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Good catch. Removed.

File: cicd/forgeops-tests/configure.py (hunk 5 above)
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: For `DEPENDENCIES`, I have renamed that.

File: cicd/forgeops-tests/configure.py (hunk 6 above)
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Line 80 => `    for component in TESTS_COMPONENTS.split(" "):`

File: cicd/forgeops-tests/configure.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: error need to fix that but need "bigger" changes

[the remark this answers is not in the document]
Comment (csovant) [reply cicd/forgeops-tests/configure.py]: Now by default the value is `""` and in that case we get the default values from `env.sh`

**Pull request merged**